src/processors/other/Tremolo.cpp

```diff
@@ -121,26 +123,46 @@ void Tremolo::processAudio (AudioBuffer<float>& buffer)
     phaseSmooth.setTargetValue (*rateParam * MathConstants<float>::pi / fs);
     waveSmooth.setTargetValue (*waveParam);
 
-    // fill wave buffer (-1, 1)
-    dsp::AudioBlock<float> waveBlock { waveBuffer };
-    dsp::ProcessContextReplacing<float> waveCtx { waveBlock };
-    fillWaveBuffer (waveBlock.getChannelPointer (0), numSamples, phase);
-
-    // shrink range to (0, 1)
-    waveBlock *= 0.5f;
-    waveBlock += 0.5f;
-
-    // apply depth parameter
-    auto depthVal = std::pow (depthParam->load(), 0.33f);
-    depthGainSmooth.setTargetValue (depthVal);
-    waveBlock.multiplyBy (depthGainSmooth);
-    depthAddSmooth.setTargetValue (1.0f - depthVal);
-    addSmoothed (waveBlock, depthAddSmooth);
-    filter.process (waveCtx);
+    if (inputsConnected.contains (1))
+    {
+        waveBuffer.copyFrom (0, 0, getInputBuffer (1), 0, 0, numSamples);
```

Comment:
Is a copyFrom here the best option, or is there a better way to transfer these samples.

Reply:
If was collapsed down to mono wouldn't that make processing easier?

Reply:
Yeah, I think collapsing to mono makes sense at least for the Tremolo module. I imagine most modulation signals will be mono, so it won't make a difference, but since it's possible to use an audio signal as a modulator, I think summing to mono makes the most sense.

Reply:
In the newest changes I stole the `makeBufferMono` lambda from stereoMerger and made it a protected member variable in `BaseProcessor`. does this make sense? Or it could stay as a lambda in both StereoMerger and Tremolo.